### DISCUSSION TITLE
Filtering with "where" based on non-unique attribute

### DISCUSSION THREAD

**Discussion by hatchli** [prisma/prisma#1971] [Q&A] [answered]
I would like to conditionally run the update resolver below _only if_ the expiry token is not expired. In Prisma 1, it was as easy as chaining comparison operators within the `where` statement. But with Prisma 2, I seemingly can't do that if the attribute is not an id or unique (like the email and the token, which are marked with @unique directives). So, how would I go about it now?

Given the following nexus-prisma mutation resolver:
```
    t.field('emailConfirmation', {
      type: 'Proposal',
      args: {
        email: stringArg({ nullable: false }),
        emailConfirmToken: stringArg({ nullable: false }),
      },
      resolve: async (
        parent,
        { email, emailConfirmToken },
        ctx,
      ) => {
        return ctx.prisma.proposal.update({
          where: { emailConfirmToken, email},
          data:{emailConfirmed: true}
        })
      },
    })
```

And the mutation generating the expiration:
```
    t.field('confirmEmail', {
      type: 'Proposal',
      args: {
        proposal_id: idArg(),
        email: stringArg({ nullable: false }),
      },
      resolve: async (parent, { email, proposal_id }, ctx) => {
        const randomBytesPromiseified = promisify(randomBytes)
        const confirmToken = (await randomBytesPromiseified(20)).toString('hex')
        const confirmTokenExpiry = Date.now() + 3600000 // 1 hour from now
        sendConfirmEmail(email, confirmToken)
        return ctx.prisma.proposal.update({
          where: { proposal_id: Number(proposal_id) },
          data: {
            emailConfirmToken: confirmToken,
            confirmTokenExpiry: Number(confirmTokenExpiry),
          },
        })
      },
    })
```

**Comment by hatchli** [comment, marked as answer]
Figured it out. You can't just go straight into update, and you also can't use findOne for some reason. For anyone in the future, use findMany.

**Reply by Sytten** [reply]
You can only use unique attributes in findOne for now, there is an issue tracking this limitation.

**Reply by hatchli** [reply]
Good stuff, thanks @Sytten, I'll look for it :)

**Comment by ludralph** [comment]
Hi there,

To keep our discussions organized and focused on the most relevant topics, we’re reviewing and tidying up our backlog. As part of this process, we’re closing discussions that have already been marked as answered but remain open.

If this discussion still requires further input or clarification, feel free to reopen it or start a new one with updated details. Your contributions are invaluable to the community, and we’re here to help!

For more details about our priorities and vision for the future of Prisma ORM, check out our latest blog post: https://www.prisma.io/blog/prisma-orm-manifesto.

Thank you for your understanding and ongoing support of the Prisma community!
